src/backend/cpu/unwrap.cpp

```diff
@@ -60,5 +60,51 @@ INSTANTIATE(uchar)
 INSTANTIATE(char)
 INSTANTIATE(short)
 INSTANTIATE(ushort)
+#undef INSTANTIATE
+
+template <typename T>
+Array<T> unwrap_dilated(const Array<T> &in, const dim_t wx, const dim_t wy,
+                        const dim_t sx, const dim_t sy, const dim_t px,
+                        const dim_t py, const dim_t dx, const dim_t dy,
+                        const bool is_column) {
+    in.eval();
+
+    af::dim4 idims = in.dims();
+
+    dim_t nx = 1 + (idims[0] + 2 * px - (((wx - 1) * dx) + 1)) / sx;
+    dim_t ny = 1 + (idims[1] + 2 * py - (((wy - 1) * dy) + 1)) / sy;
+
+    af::dim4 odims(wx * wy, nx * ny, idims[2], idims[3]);
+
+    if (!is_column) { std::swap(odims[0], odims[1]); }
+
+    Array<T> outArray = createEmptyArray<T>(odims);
+
+    const int d = (is_column) ? 1 : 0;
+    getQueue().enqueue(kernel::unwrap_dim_dilated<T>, outArray, in, wx, wy, sx,
+                       sy, px, py, dx, dy, d);
+
+    return outArray;
+}
+
+#define INSTANTIATE(T)                                                      \
+    template Array<T> unwrap_dilated<T>(                                    \
+        const Array<T> &in, const dim_t wx, const dim_t wy, const dim_t sx, \
+        const dim_t sy, const dim_t px, const dim_t py, const dim_t dx,     \
+        const dim_t dy, const bool is_column);
+
+INSTANTIATE(float)
+INSTANTIATE(double)
+INSTANTIATE(cfloat)
+INSTANTIATE(cdouble)
+INSTANTIATE(int)
+INSTANTIATE(uint)
+INSTANTIATE(intl)
+INSTANTIATE(uintl)
+INSTANTIATE(uchar)
+INSTANTIATE(char)
+INSTANTIATE(short)
+INSTANTIATE(ushort)
+#undef INSTANTIATE
```

Comment:
I think we can remove the instantiations for cfloat, cdouble , intl and uintl since I removed the those instantiations in strided convolution and convolve2Gradient @syurkevi Can you please confirm

Reply:
I think its fine to have this function instantiated for those types. Its more generic than convolve.

Reply:
We would be only increasing our compilation times without any added benefits since `unwrap_dilated` is not exposed to the user at the moment. `unwrap_dilated` is only used internally by the new convolution functions that were added in this PR. All the other instantiations except `float` & `double` are basically not used at all anywhere else in the code. These extra instantiations can be added later if and when required.